source/kraken/kraken_zmq.h

```diff
@@ -91,6 +91,7 @@ inline void doWork(zmq::context_t& context,
             continue;
         }
 
+        auto in_flight_guard = metrics.start_in_flight();
```

Comment:
minor: I think you can construct an object InFlightGuard here without unique_ptr?

Reply:
should be better now